src/lib.rs

```diff
@@ -12,6 +12,186 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Rust client guide
```

Comment:
What is the advantage of doing this in `lib.rs` vs README.md?

Reply:
The only one would be that it will be added to the crate documentation when publishing. The readme could link to the documentation. Do you have any preferences?

Reply:
@marot is it possible to have the README be the "source of truth" and then have that added procedurally to the crate documentation? If that's too much fuss or not possible then having the readme link to the documentation sounds okay.

Reply:
Unfortunately, it is not too easy, because the examples contain code that is supposed to be hidden (setup code).

I will link to the documentation for now.